afqinsight/nn/performance_utils.py

```diff
@@ -0,0 +1,328 @@
+import tempfile
+
+import numpy as np
+import tensorflow as tf
+import torch
+from neurocombat_sklearn import CombatModel
+from sklearn.impute import SimpleImputer
+from sklearn.model_selection import train_test_split
+
+from afqinsight import AFQDataset
+
+
+def prep_tensorflow_data():
+    dataset = AFQDataset.from_study("hbn")
+    dataset.drop_target_na()
+
+    # converts the input array to a tensor, fit for training
+    def array_to_tensor(input_array):
+        return np.transpose(
+            np.array([input_array[:, i * 100 : (i + 1) * 100] for i in range(48)]),
```

Comment:
The numbers 48 and 100 are here a "magic number". They're tied here to the number of tracts and metrics that are used in HBN, as well as the number of nodes, but could be something else for other datasets. Consider how you automatically derive these numbers from the dataset.